cabal-install/Distribution/Solver/Modular/Tree.hs

```diff
@@ -37,17 +37,17 @@ type Weight = Double
 -- package, flag, and stanza choices control the traversal order.
 -- TODO: The weight type should be changed from [Double] to Double to avoid
 -- giving too much weight to preferences that are applied later.
-data Tree a =
+data Tree a b =
```

Comment:
One way to solve this is to point out a few places where we specialize, and describe what the nodes are. Commentary about which type of node the value is stored on also seems helpful; you could even name it `done`, `choice` e.g. to give readers the right idea.